servers/visual_server.cpp

```diff
@@ -474,14 +503,13 @@ Error VisualServer::_surface_set_data(Array p_arrays, uint32_t p_format, uint32_
 
 					for (int i = 0; i < p_vertex_array_len; i++) {
 
-						int8_t vector[4] = {
-							(int8_t)CLAMP(src[i].x * 127, -128, 127),
-							(int8_t)CLAMP(src[i].y * 127, -128, 127),
-							(int8_t)CLAMP(src[i].z * 127, -128, 127),
-							0,
+						Vector2 res = norm_to_oct(src[i]);
+						int8_t vector[2] = {
+							(int8_t)CLAMP(res.x * 127, -128, 127),
```

Comment:
Might want to look into rounding these values since it might improve quality slightly